apps/api-nexus/src/app/lib/bullMQ/bullMQ.service.ts

```diff
@@ -0,0 +1,269 @@
+    refreshToken: string;
+  };
+}
+
+@Injectable()
+export class BullMQService {
+  constructor(
+    private readonly prismaService: PrismaService,
+    @InjectQueue("nexus-batch-worker") private readonly batchJobQueue: Queue
+  ) {}
+
+  private async createUploadResourceBatchJob(
+    accessToken: string,
+    batchId: string,
+    channel: Channel
+  ): Promise<Array<Bull.Job<unknown>>> {
+    const batch = await this.prismaService.batch.findUnique({
+      where: {
+        id: batchId,
+      },
+      include: {
+        batchTasks: true,
+      },
+    });
+
+    if (batch == null) {
+      throw new Error("Batch not found @ worker");
+    }
+
+    const channelData = await this.prismaService.channel.findUnique({
+      where: { id: channel.id },
+    });
+
+    const jobs = await Promise.all(
+      batch.batchTasks.map(async (batchTask) => {
+        const task = batchTask.task as {
+          channelId: string;
+          resourceId: string;
+        };
+        const resource = await this.prismaService.resource.findUnique({
+          where: { id: task.resourceId ?? "" },
+          include: {
+            resourceLocalizations: true,
+          },
+        });
+
+        if (resource == null) {
+          throw new Error("Resource not found");
+        }
+
+        const jobData: UploadResourceJob = {
+          batchId: batch.id,
+          batchTaskId: batchTask.id,
+          accessToken,
+          resource: {
+            id: resource.id,
+            fileUrl: resource.name,
+            title: resource.resourceLocalizations[0]?.title ?? "",
+            description: resource.resourceLocalizations[0]?.description ?? "",
+            thumbnailFileUrl: resource?.customThumbnail ?? "",
+            captionFileUrl:
+              resource?.resourceLocalizations[0]?.captionFile ?? "",
+            privacyStatus: resource.privacy ?? PrivacyStatus.private,
+          },
+          channel: {
+            id: channel?.id ?? "",
+            channelId: channelData?.youtubeId ?? "",
+          },
+        };
+
+        return {
+          name: batchTask.type,
+          data: jobData,
+        };
+      })
+    );
+    return await this.batchJobQueue.addBulk(
+      jobs.filter((job) => job.data !== undefined)
+    );
+  }
+
+  async createUploadResourceBatch(
+    token: string,
+    batchName: string,
+    channel: Channel,
+    resources: Resource[]
+  ): Promise<Batch> {
+    const batch = await this.prismaService.batch.create({
+      data: {
+        name: batchName,
+      },
+    });
+    await this.prismaService.batchTask.createMany({
+      data: resources.map((resource) => {
+        return {
+          batchId: batch.id,
+          type: "processResourceUpload",
+          task: {
+            resourceId: resource.id,
+            channelId: channel.id,
+          },
+        };
+      }),
+    });
+
+    await this.createUploadResourceBatchJob(token, batch.id, channel);
+
+    return batch as unknown as Batch;
```

Comment:
type should be correct.

```suggestion
    return batch;
```